test/validation/api/scheduler/scheduler.c

```diff
@@ -1982,6 +1982,87 @@ static void parallel_execute(odp_schedule_sync_t sync, int num_queues,
 	reset_queues(args);
 }
 
+/* Worker creates a private schedule group and a queue in it, enqueues events,
+ * and verifies scheduled events come only from that private group. */
+static int private_group_thread(void *arg)
+{
+	thread_args_t *thr_args = arg;
+	test_globals_t *globals = thr_args->globals;
+	odp_pool_t pool = odp_pool_lookup(MSG_POOL_NAME);
+	odp_thrmask_t mask;
+	odp_schedule_group_t private_group;
+	odp_queue_param_t queue_param;
+	odp_queue_t enqueue_to, sched_from;
+	odp_event_t ev;
+	odp_buffer_t buffer;
+	uint32_t i;
+	const uint32_t num_ev = 16;
+	int thr_id;
+	char group_name[ODP_SCHED_GROUP_NAME_LEN];
+	char queue_name[ODP_QUEUE_NAME_LEN];
+
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	thr_id = odp_thread_id();
+	snprintf(group_name, sizeof(group_name), "pgt_group_%d", thr_id);
+	snprintf(queue_name, sizeof(queue_name), "pgt_queue_%d", thr_id);
+
+	odp_thrmask_zero(&mask);
+	odp_thrmask_set(&mask, thr_id);
+
+	/* Create private group for this worker (only this thread in the mask) */
+	private_group = odp_schedule_group_create(group_name, &mask);
+	CU_ASSERT_FATAL(private_group != ODP_SCHED_GROUP_INVALID);
+
+	odp_queue_param_init(&queue_param);
+	queue_param.type        = ODP_QUEUE_TYPE_SCHED;
+	queue_param.sched.sync  = ODP_SCHED_SYNC_PARALLEL;
+	queue_param.sched.group = private_group;
+	queue_param.size        = num_ev;
+
+	enqueue_to = odp_queue_create(queue_name, &queue_param);
+	CU_ASSERT_FATAL(enqueue_to != ODP_QUEUE_INVALID);
```

Comment:
To make the test potentially more complex for the implementation, I'd create the schedule groups and queues, and fill the queues already in the control thread. Workers would then only join their specific group.